snap/info.go

```diff
@@ -82,7 +82,12 @@ func MinimalPlaceInfo(name string, revision Revision) PlaceInfo {
 
 // MountDir returns the base directory where it gets mounted of the snap with the given name and revision.
 func MountDir(name string, revision Revision) string {
-	return filepath.Join(dirs.SnapMountDir, name, revision.String())
+	return MountDirWithBasePath(dirs.SnapMountDir, name, revision)
+}
+
+// MountDirWithBasePath returns the base directory where it gets mounted of the snap with the given name and revision.
+func MountDirWithBasePath(basePath, name string, revision Revision) string {
```

Comment:
Let's please drop this function. We don't need a special (and long) name for filepath.Join. :-)